wolfcrypt/src/evp.c

```diff
@@ -6803,6 +6805,870 @@ void wolfSSL_EVP_PKEY_free(WOLFSSL_EVP_PKEY* key)
+            return WOLFSSL_FAILURE;
+
+        if (GetSequence(pkey, &inOutIdx, &length, pkeySz) < 0)
+            return WOLFSSL_FAILURE;
+    }
+    /* Get modulus */
+    if (GetASNTag(pkey, &inOutIdx, &tag, pkeySz ) < 0)
+        return WOLFSSL_FAILURE;
+
+    if (tag != ASN_INTEGER)
+        return WOLFSSL_FAILURE;
+
+    if (GetLength(pkey, &inOutIdx, &length, pkeySz) < 0)
+        return WOLFSSL_FAILURE;
+
+    nSz = length;
+    n   = (byte*)(&pkey[inOutIdx]);
+    inOutIdx += length;
+
+    /* Get exponent */
+
+    if (GetASNTag(pkey, &inOutIdx, &tag, pkeySz) < 0)
+        return WOLFSSL_FAILURE;
+
+    if (tag != ASN_INTEGER)
+        return WOLFSSL_FAILURE;
+
+    if (GetLength(pkey, &inOutIdx, &length, pkeySz) < 0)
+        return WOLFSSL_FAILURE;
+
+
+    eSz = length;
+    e   = (byte*)(&pkey[inOutIdx]);
+
+    /* print out public key elements */
+    idx = 0;
+    XMEMSET(buff, 0, sizeof(buff));
+    Indent(out, indent);
+    if (wolfSSL_BIO_write(out, "RSA Public-Key: (",
+                          sizeof("RSA Public-Key: (") -1) < 0)
+        return WOLFSSL_FAILURE;
+
+    if (mp_set_int(&a, bitlen) != 0)
+        return WOLFSSL_FAILURE;
+    if (mp_todecimal(&a, (char*)buff) != 0)
+        return WOLFSSL_FAILURE;
+    wsz = (int)XSTRLEN((const char*)buff);
+
+    if (wolfSSL_BIO_write(out, buff + idx, wsz) < 0)
+        return WOLFSSL_FAILURE;
+
+    if (wolfSSL_BIO_write(out, " bit)\n", sizeof(" bit)\n") -1) < 0)
+        return WOLFSSL_FAILURE;
+
+    /* print Modulus */
+    Indent(out, indent);
+    if (wolfSSL_BIO_write(out, "Modulus:\n", sizeof("Modulus:\n") -1) < 0)
+        return WOLFSSL_FAILURE;
+
+    if (DumpElement(out, n, nSz, indent + 4) != WOLFSSL_SUCCESS)
+        return WOLFSSL_FAILURE;
+
+    /* print public Exponent */
+    idx = 0;
+    Indent(out, indent);
+    if (wolfSSL_BIO_write(out, "Exponent: ", sizeof("Exponent: ") -1) < 0)
+        return WOLFSSL_FAILURE;
+
+    for (i = 0; i < eSz; i++) {
+        exponent <<= 8;
+        exponent += e[i];
+    }
+
+    XMEMSET(buff, 0, sizeof(buff));
+    if (mp_set_int(&a, exponent) != 0)
+        return WOLFSSL_FAILURE;
+    if (mp_todecimal(&a, (char*)buff) != 0)
+        return WOLFSSL_FAILURE;
+    wsz = (int)XSTRLEN((const char*)buff);
+
+    if (wolfSSL_BIO_write(out, buff + idx, wsz) < 0)
+        return WOLFSSL_FAILURE;
+
+    if (wolfSSL_BIO_write(out, " (0x", sizeof(" (0x") -1) < 0)
+        return WOLFSSL_FAILURE;
+
+    idx = 0;
+    XMEMSET(buff, 0, sizeof(buff));
+    for (i = 0; i < eSz; i++) {
+        outSz = sizeof(outHex);
+        if (Base16_Encode((const byte*)&e[i], 1, outHex, &outSz ) != 0)
+            return WOLFSSL_FAILURE;
+
+        if ((unsigned int)idx + 2 <= sizeof(buff)) {
+            XMEMCPY(buff + idx, outHex, 2);
+            idx += 2;
+        }
+    }
+
+    if (wolfSSL_BIO_write(out, buff, idx) < 0)
+        return WOLFSSL_FAILURE;
+
+    if (wolfSSL_BIO_write(out, ")\n", sizeof(")\n") -1) < 0)
```

Comment:
Lets use XSTRLEN instead of sizeof - 1, if wanting to not have the null terminator here and throughout the function with the calls to wolfSSL_BIO_write

Reply:
Replaced with XSTRLEN.